curator-recipes/src/test/java/org/apache/curator/framework/recipes/shared/TestSharedCount.java

```diff
@@ -389,4 +390,93 @@ public void processResult(CuratorFramework client, CuratorEvent event) throws Ex
             CloseableUtils.closeQuietly(curatorFramework);
         }
     }
+
+    @Test
+    public void testDisconnectReconnectEventDoesNotFireValueWatcherWithMultipleClients() throws Exception
```

Comment:
I don't think this tests the problem. I tried this test with an unchanged version SharedValue and it passes.